temporalio/worker/_workflow.py

```diff
@@ -193,6 +206,48 @@ async def _handle_activation(
     ) -> None:
         global LOG_PROTOS
 
+        def get_traceback_for_thread(
+            thread_id: int,
+        ) -> Tuple[Optional[TracebackType], bool]:
+            """Take a thread id and attempt to construct a traceback from its frames.
+
+            Args:
+                thread_id: Thread ID to be checked.
+
+            Returns:
+                A tuple containing:
+                    0: A TracebackType if it was able to be constructed with a valid TID, None otherwise.
+                    1: A bool that determines whether the error message should warn about a possibly incomplete traceback.
+            """
+            # retrieve all frames
+            frames = sys._current_frames()
+
+            # if the id is not present, the thread must have terminated or there is another error
+            if thread_id not in frames.keys():
+                return (None, False)
+
+            frame = sys._current_frames()[thread_id]
+            top_tb = TracebackType(
+                None, tb_frame=frame, tb_lasti=frame.f_lasti, tb_lineno=frame.f_lineno
+            )
+
+            # Construct tb stack, might be a better way/builtin to do this
+            tb = top_tb
+            try:
+                while frame.f_back:
+                    frame = frame.f_back
+                    tb.tb_next = TracebackType(
+                        None,
+                        tb_frame=frame,
+                        tb_lasti=frame.f_lasti,
+                        tb_lineno=frame.f_lineno,
+                    )
+                    tb = tb.tb_next
+            except Exception:
```

Comment:
OK so as your previous comment said, catching `Exception` is questionable and should be commented. What are examples of exceptions we're catching here? Also if exception-catching is needed I would prefer to do it inside the while loop, i.e. so that the `try` block is as narrowly-scoped as possible around the code (i.e. line) where the exception is expected.